packages/docusaurus-theme-classic/src/theme/hooks/useWindowSize.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {useEffect, useState} from 'react';
+
+const desktopThresholdWidth = 996;
+const desktopSize = 'desktop';
+const mobileSize = 'mobile';
+
+function useWindowSize() {
+  const isClient = typeof window === 'object';
+
+  function getSize() {
+    return isClient
+      ? (window.innerWidth > desktopThresholdWidth && desktopSize) || mobileSize
+      : undefined;
```

Comment:
I'd suggest extracting the fn outside of the hook + making it more "readable". 

Usage of `&&` and  `||` is hard to follow


```
function getSize() {
  if (isClient) {
    return undefined;
  }
  returrn window.innerWidth > desktopThresholdWidth ? desktopSize : mobileSize;
}
```

Reply:
Indeed, much clearer. I updated the code.